tests/test_documents.py

```diff
@@ -161,3 +204,59 @@ async def test_add_document_ingest_error(client):
         client=client, collection=collection.name, document_id=document.id
     )
     assert len(chunks) == 0
+
+
+async def test_upload_document_unknown_document_id(client, doc_fixture):
+    response = await upload_document_content.asyncio_detailed(
+        client=client,
+        document_id=1_000_000,
+        body=BodyUploadDocumentContent(
+            content=File(
+                payload=NEARLY_EMPTY_BYTES,
+                file_name="file-name-1",
+                mime_type="application/pdf",
+            ),
+        ),
+    )
+    assert response.status_code == 404
+
+
+async def test_document_lifecycle(client, doc_fixture):
+    # 1. Upload a PDF for one of the fixutre docs and verify the document is "pending"
+    await upload_test_pdf(client, doc_fixture.doc1, NEARLY_EMPTY_BYTES)
+
+    # 2. Wait for ingestion to complete (would be nicer if we could hook into the queue somehow)
+    # and verify the PDF has been ingested correctly
+    await document_ingested(client, doc_fixture.doc1)
+
+    document2 = await get_document.asyncio(doc_fixture.doc1, client=client)
+    assert document2
+    assert document2.id == doc_fixture.doc1
+    assert document2.extracted_text == NEARLY_EMPTY_TEXT
+    assert document2.url is None
+    assert document2.ingest_state == IngestState.INGESTED
+    assert document2.ingest_error is None
+
+    chunks = await list_chunks.asyncio(client=client, document_id=doc_fixture.doc1)
+    assert chunks
+
+    # 3. Upload a revised PDF and verify the document is back into "pending" state
+    await upload_test_pdf(client, doc_fixture.doc1, NEARLY_EMPTY_BYTES2)
+
+    # 4. Wait for the new doc to be ingested and verify it was ingested correctly
+    await document_ingested(client, doc_fixture.doc1)
```

Comment:
Cool. Can probably use these elsewhere in this test and test_e2e to make it more maintainable when we change things up.